core/mixins.py

```diff
@@ -297,6 +298,16 @@ def get_visibilite_display_text(self) -> str:
             case Visibilite.NATIONALE:
                 return "Toutes les structures"
 
+    def update_allowed_structures_and_visibility(self, contacts: ContactQueryset):
+        if not contacts:
+            return
+        if not self.is_visibilite_nationale:
+            structures = [contact.structure if contact.structure else contact.agent.structure for contact in contacts]
```

Comment:
```suggestion
            structures = [contact.structure or contact.agent.structure for contact in contacts]
```

Cela donnerait la même chose non ?